src/Juvix/Compiler/Core/Transformation/CheckExec.hs

```diff
@@ -0,0 +1,40 @@
+module Juvix.Compiler.Core.Transformation.CheckExec where
+
+import Data.HashMap.Strict qualified as HashMap
+import Juvix.Compiler.Core.Error
+import Juvix.Compiler.Core.Extra
+import Juvix.Compiler.Core.Transformation.Base
+import Juvix.Data.PPOutput
+
+checkExec :: forall r. Member (Error CoreError) r => InfoTable -> Sem r InfoTable
+checkExec tab =
+  case tab ^. infoMain of
+    Nothing -> return tab
+    Just sym ->
+      case ii ^. identifierType of
+        NPi {} ->
+          throw
+            CoreError
+              { _coreErrorMsg = ppOutput "`main` cannot have a function type for this target",
+                _coreErrorNode = Nothing,
+                _coreErrorLoc = loc
+              }
+        ty
+          | isTypeConstr tab ty ->
+              throw
+                CoreError
+                  { _coreErrorMsg = ppOutput "`main` cannot be a type for this target",
+                    _coreErrorNode = Nothing,
+                    _coreErrorLoc = loc
+                  }
+        _ ->
+          return tab
+      where
+        ii = fromJust $ HashMap.lookup sym (tab ^. infoIdentifiers)
+        loc = fromMaybe defaultLoc (ii ^. identifierLocation)
+
+        mockFile :: Path Abs File
+        mockFile = $(mkAbsFile "/core-to-geb")
```

Comment:
Suggestion: Does the mockFile name make sense here, given that the transform is not used in the Geb pipeline?

I'm not sure what to suggest instead. Something like `/check-exec` might help identify where the error is coming from.

Reply:
Yes, I just copied it and forgot to change the name